test/functional/lua/interrupts_spec.lua

```diff
@@ -187,6 +192,14 @@ describe(':luado command', function()
     feed(':cq<CR>')
     screen_check_exit(screen)
   end)
+  it('may be interrupted with <C-c> without additional :terminal layer',
+  function()
+    feed('iabc<CR>def<CR>ghi<Esc>:%luado while true do i = nil end<CR>')
+    sleep(100)
+    feed('<C-c><C-c><C-c>')
```

Comment:
`:luado` executes lua function once for each line, others execute it only once. Specifically for this test (unlike the above `:luado` one) three should be not needed because first one should yield E5111 though, but I did not check, in any case there will likely be more tests refactoring.